pkg/reader/usdc_reader.go

```diff
@@ -0,0 +1,308 @@
+package reader
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+
+	sel "github.com/smartcontractkit/chain-selectors"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query"
+
+	"github.com/smartcontractkit/chainlink-ccip/execute/exectypes"
+	"github.com/smartcontractkit/chainlink-ccip/pkg/consts"
+	"github.com/smartcontractkit/chainlink-ccip/pluginconfig"
+)
+
+type MessageHash []byte
+
+type USDCMessageReader interface {
+	MessageHashes(ctx context.Context,
+		source, dest cciptypes.ChainSelector,
+		tokens map[exectypes.MessageTokenID]cciptypes.RampTokenAmount,
+	) (map[exectypes.MessageTokenID]MessageHash, error)
+}
+
+const (
+	CCTPMessageVersion = uint32(0)
+)
+
+// TODO, this should be fetched from USDC Token Pool and cached
+var CCTPDestDomains = map[uint64]uint32{
+	sel.ETHEREUM_MAINNET.Selector:            0,
+	sel.AVALANCHE_MAINNET.Selector:           1,
+	sel.ETHEREUM_MAINNET_OPTIMISM_1.Selector: 2,
+	sel.ETHEREUM_MAINNET_ARBITRUM_1.Selector: 3,
+	sel.ETHEREUM_MAINNET_BASE_1.Selector:     6,
+	sel.POLYGON_MAINNET.Selector:             7,
+	sel.AVALANCHE_TESTNET_FUJI.Selector:      1,
+}
+
+type usdcMessageReader struct {
+	contractReaders map[cciptypes.ChainSelector]types.ContractReader
```

Comment:
```suggestion
	contractReaders map[cciptypes.ChainSelector]contractreader.ContractReaderFacade
```

Reply:
I'm curious what @asoliman92  things, but at this stage I would prefer we avoid `types.ContractReader` everywhere except for `CCIPReader`. Especially with the new `DiscoveryProcessor` I think it will make things a lot simpler.

Reply:
There is one major issue with `ContractReaderFacade`, it's in the internal package, I can't use that in the integration tests in the Chainlink repository.  Also, we don't need that facade to be used there, because we can rely on the pure interfaces. DiscoveryProcessor is not going to be a thing for USDCReader, because contracts' addresses are passed using the config.

I've added missing methods to the `ContractReaderFacade` and it can be used interchangeably with the `ContractReader`. Mocking works without any additional magic

Reply:
> I would prefer we avoid types.ContractReader everywhere except for CCIPReader.

What's the reasoning behind this?